### DISCUSSION TITLE
I am no longer able to successfully deploy to vercel.

### DISCUSSION THREAD

**Discussion by LarryEitel** [vercel/community#2963] [Help]
### Summary

I am getting

```
No Output Directory named "public" found after the Build completed. You can configure the Output Directory in your Project Settings.
```

In the build log:

```
Error: No Output Directory named "public" found after the Build completed. You can configure the Output Directory in your Project Settings.
Learn More: https://vercel.link/missing-public-directory
STATIC_BUILD_NO_OUT_DIR: No Output Directory named "public" found after the Build completed. You can configure the Output Directory in your Project Settings.
```

It was able to deploy about a week ago. I don't see any obvious differences aside from various package upgrades.

Thoughts please...

### Example

https://discord.com/channels/457912077277855764/1121547244441710633

### Steps to Reproduce

https://discord.com/channels/457912077277855764/1121547244441710633

**Comment by amyegan** [comment]
It sounds like you have "public" configured as the [Output Directory](https://vercel.com/docs/concepts/deployments/configure-a-build#output-directory) for your project, but no "public" directory was found after the build step completed. Try changing the output directory value to match the expected output directory used for your project.

**Comment by LarryEitel** [comment]
Thank you Amy, Should I continue on this here or via Discord? 

What I have done:

I changed Output Directory on settings to: '.svelte-kit/output'. With that I am able to build and deploy. However, the URL returns 404.

**Comment by LarryEitel** [comment]
I just opened a "Pro Case" 00142990 on this as well.

**Comment by amyegan** [comment]
Thanks for letting me know about the support case. If you're having a good discussion on Discord then please feel free to continue there. But the folks there are more likely to be focused on Svelte rather than Vercel-specific features and configuration. To reach the team focused on Vercel support, I'd recommend following up with your Pro case or staying with this discussion. 

Are you using the `SvelteKit` preset under Build & Development Settings? If not, please try that and let me know how it goes (or share with your open case). 

If that doesn't work, it would be very helpful to get your public repo or a minimal reproduction if you are able to share it either here or in your support case. That is usually the fastest way for us to replicate a problem and figure out what is going wrong.

**Comment by LarryEitel** [comment]
Ok, so...
I successfully deployed and can update and redeploy a vanilla project at:
dev.pinz.ai
https://vercel.com/larry-eitel-s-team/pinz-dev

However, I am still unable to deploy an update for project pinz:

https://vercel.com/larry-eitel-s-team/pinz/FpYH629Ho1x5Wmeksr9gtjg5ED72

```
Error: No Output Directory named "public" found after the Build completed. You can configure the Output Directory in your Project Settings.
Learn More: https://vercel.link/missing-public-directory
STATIC_BUILD_NO_OUT_DIR: No Output Directory named "public" found after the Build completed. You can configure the Output Directory in your Project Settings.
```

Amy I urgently need to get this deployed before the weekend.

**Comment by LarryEitel** [comment]
Next thing is that I deleted .vercel from root of my pinz project. And then ran Vercel CLI:

```
$ vercel
Vercel CLI 30.2.3
? Set up and deploy “C:\__repos\_wip\pinz.ai”? [Y/n] y
? Which scope do you want to deploy to? Larry Eitel's Team
? Link to existing project? [y/N] y
? What’s the name of your existing project? pinz
🔗  Linked to larry-eitel-s-team/pinz (created .vercel)
🔍  Inspect: https://vercel.com/larry-eitel-s-team/pinz/FK6mxujAXQNyGFATjNaVeHK2ijyA [7s]
Error: No Output Directory named "public" found after the Build completed. You can configure the Output Directory in your Project Settings.
```

**Comment by LarryEitel** [comment]
Here is my package.json

```
{
	"name": "pinz.ai",
	"version": "0.0.2",
	"private": true,
	"scripts": {
		"dev": "vite dev",
		"build": "vite build",
		"preview": "vite preview",
		"check": "svelte-kit sync && svelte-check --tsconfig ./tsconfig.json",
		"check:watch": "svelte-kit sync && svelte-check --tsconfig ./tsconfig.json --watch",
		"lint": "prettier --plugin-search-dir . --check . && eslint .",
		"format": "prettier --plugin-search-dir . --write .",
		"test": "vitest",
		"coverage": "vitest run --coverage",
		"test:jest": "jest --config jest.config.cjs  --experimental-vm-modules ./node_modules/.bin/jest",
		"test:unit": "vitest"
	},
	"devDependencies": {
		"@auth/core": "^0.8.4",
		"@auth/sveltekit": "^0.3.3",
		"@googlemaps/js-api-loader": "^1.16.2",
		"@iconify/svelte": "^3.1.4",
		"@skeletonlabs/skeleton": "^1.8.0",
		"@supabase/auth-helpers-sveltekit": "^0.10.1",
		"@supabase/supabase-js": "^2.26.0",
		"@sveltejs/adapter-auto": "^2.1.0",
		"@sveltejs/adapter-vercel": "^3.0.1",
		"@sveltejs/kit": "^1.20.4",
		"@sveltejs/vite-plugin-svelte": "^2.4.1",
		"@tailwindcss/forms": "^0.5.3",
		"@tailwindcss/line-clamp": "^0.4.4",
		"@tailwindcss/typography": "^0.5.9",
		"@types/cookie": "^0.5.1",
		"@types/debounce": "^1.2.1",
		"@types/google.maps": "^3.53.4",
		"@types/luxon": "^3.3.0",
		"@types/node": "^20.3.1",
		"@types/pg": "^8.10.2",
		"@typescript-eslint/eslint-plugin": "^5.60.0",
		"@typescript-eslint/parser": "^5.60.0",
		"@vercel/analytics": "^1.0.1",
		"chalk": "^5.2.0",
		"class-variance-authority": "^0.6.0",
		"cookie": "^0.5.0",
		"cors": "^2.8.5",
		"date-fns": "^2.30.0",
		"dayjs": "^1.11.8",
		"debounce": "^1.2.1",
		"delay": "^5.0.0",
		"drizzle-kit": "^0.18.1",
		"drizzle-orm": "^0.26.5",
		"eslint": "^8.43.0",
		"eslint-config-prettier": "^8.8.0",
		"eslint-plugin-svelte3": "^4.0.0",
		"felte": "^1.2.10",
		"luxon": "^3.3.0",
		"pg": "^8.11.0",
		"postcss": "^8.4.24",
		"postcss-load-config": "^4.0.1",
		"prettier": "^2.8.8",
		"prettier-plugin-svelte": "^2.10.1",
		"svelte": "^3.59.2",
		"svelte-check": "^3.4.4",
		"svelte-french-toast": "1.0.4-beta.0",
		"svelte-i18n": "^3.6.0",
		"svelte-preprocess": "^5.0.4",
		"tailwindcss": "^3.3.2",
		"ts-node": "^10.9.1",
		"tslib": "^2.5.3",
		"typescript": "^5.1.3",
		"unplugin-icons": "^0.16.3",
		"vite": "^4.3.9",
		"vitest": "^0.32.2",
		"xss": "^1.0.14",
		"zod": "^3.21.4",
		"zodock": "^0.0.12"
	},
	"type": "module",
	"dependencies": {
		"autoprefixer": "^10.4.14"
	}
}

```

**Comment by amyegan** [comment]
I was also able to deploy a simple boilerplate SvelteKit project, so I think this problem is specific to the pinz project. The next thing I can think to try is verifying that the build command is set correctly for the project.

Another idea is to try running the build command locally to check the name of the generated output directory. Maybe something in the project configuration is causing it to use a non-standard name.

**Comment by LarryEitel** [comment]
I have narrowed the problem down to the preprocess block:

```
import adapter from "@sveltejs/adapter-vercel";
// import adapter from "@sveltejs/adapter-auto";
import { vitePreprocess } from '@sveltejs/kit/vite';


/** @type {import('@sveltejs/kit').Config} */
const config = {
	adapter: adapter(),
	files: {
		assets: 'static',
	},
	preprocess: [
		vitePreprocess({
			postcss: true
		})
	]
};

export default config;
```

I have been adding stuff from the problem repo and it deploys unless I remove the preprocess block. Continuing to explore...

**Comment by LarryEitel** [comment]
my project depends on this block:

```
	preprocess: [
		vitePreprocess({
			postcss: true
		})
	]
```

I don't understand how I can work around that at this point. Will continue exploring.

Off hand, I think it ran with that block previously.

**Comment by LarryEitel** [comment]
I can build using: pnpm build with no problem.

Should I be getting the following:

```
No adapter specified
See https://kit.svelte.dev/docs/adapters to learn how to configure your app to run on the platform of your choosing
```

**Comment by LarryEitel** [comment]
The following svelte.config builds and deploys:

```
import adapter from '@sveltejs/adapter-auto';
import {vitePreprocess} from '@sveltejs/kit/vite';

/** @type {import('@sveltejs/kit').Config} */
const config = {
    kit: {
        // adapter-auto only supports some environments, see https://kit.svelte.dev/docs/adapter-auto for a list.
        // If your environment is not supported or you settled on a specific environment, switch out the adapter.
        // See https://kit.svelte.dev/docs/adapters for more information about adapters.
        adapter: adapter(),
        files: {
            assets: 'static',
        },
    }
};

export default config;
```

However, when I use the following I get the error that I have been getting:

```
import { vitePreprocess } from '@sveltejs/kit/vite';
import adapter from "@sveltejs/adapter-auto";


/** @type {import('@sveltejs/kit').Config} */
const config = {
  adapter: adapter(),
  files: {
    assets: 'static',
  },
  preprocess: [
    vitePreprocess({
      postcss: true
    })
  ]
};

export default config;
``


Returns error:
```
$ vercel --prod
Vercel CLI 30.2.3
🔍  Inspect: https://vercel.com/larry-eitel-s-team/pinz-dev/7j9UbZ5dp7xVnsTCNwMuUp3dFznP [3s]
Error: No Output Directory named "public" found after the Build completed. You can configure the Output Directory in your Project Settings.
```

**Comment by LarryEitel** [comment]
Ok, I was able resolve it!!! It was a malformed svelte.config.js!!! This one works:

```
import { vitePreprocess } from '@sveltejs/kit/vite';
import adapter from '@sveltejs/adapter-auto';

/** @type {import('@sveltejs/kit').Config} */
const config = {
  kit: {
    adapter: adapter()
  },
  files: {
    assets: 'static',
  },
  preprocess: [
    vitePreprocess({
      postcss: true
    })
  ]
};

export default config;
```
